### PR TITLE
Map related-to null type for H2 related works and links

### PR DESCRIPTION
**Pull request by mjgiarlo**
## Why was this change made?

Connects to https://app.honeybadger.io/projects/50568/faults/69893421 which is causing all deposits to fail in stage.

## How was this change tested?

CI

## Which documentation and/or configurations were updated?

None

### CODE REVIEW

File: app/services/cocina/to_fedora/descriptive/related_resource.rb [hunk 1]
```diff
@@ -33,7 +33,8 @@ def initialize(xml:, related_resources:)
         def write
           Array(related_resources).each do |related|
             attributes = {}
-            attributes[:type] = TYPES.fetch(related.type) if related.type
+            # NOTE: 'related to' is a null type, in effect
```
Comment (jcoyne) [inline app/services/cocina/to_fedora/descriptive/related_resource.rb:36]: Could we add a 'related to' type into the map that returns a nil? I feel like this should be in that mapping.  Then call `attributes.compact` on line 39.

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/services/cocina/to_fedora/descriptive/related_resource.rb (hunk 1 above)
Comment (mjgiarlo) [reply app/services/cocina/to_fedora/descriptive/related_resource.rb:36]: sure.

File: spec/services/cocina/to_fedora/descriptive/related_resource_spec.rb [hunk 2]
```diff
@@ -117,6 +117,71 @@
     end
   end
 
+  context 'when it has a related item with the generic "related to" type (related link from H2)' do
```
Comment (ndushay) [inline spec/services/cocina/to_fedora/descriptive/related_resource_spec.rb:120]: is the the same as the next context statement, or am I missing the difference?

File: app/services/cocina/to_fedora/descriptive/related_resource.rb (hunk 1 above)
Comment (mjgiarlo) [reply app/services/cocina/to_fedora/descriptive/related_resource.rb]: done. take another look?

File: spec/services/cocina/to_fedora/descriptive/related_resource_spec.rb (hunk 2 above)
Comment (mjgiarlo) [reply spec/services/cocina/to_fedora/descriptive/related_resource_spec.rb:120]: not the same. just very similar: related link vs. related work

**Review by jcoyne — APPROVED** [review]

**Pull request merged**